runelite-client/src/test/java/net/runelite/client/plugins/opponentinfo/OpponentInfoPluginTest.java

```diff
@@ -0,0 +1,199 @@
+package net.runelite.client.plugins.opponentinfo;
+
+import com.google.inject.Guice;
+import com.google.inject.Inject;
+import com.google.inject.testing.fieldbinder.Bind;
+import com.google.inject.testing.fieldbinder.BoundFieldModule;
+import net.runelite.api.*;
+import net.runelite.api.events.InteractingChanged;
+import net.runelite.client.ui.overlay.OverlayManager;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class OpponentInfoPluginTest
+{
+    @Mock
+    @Bind
+    Client client;
+
+    @Mock
+    @Bind
+    OverlayManager overlayManager;
+
+    @Mock
+    @Bind
+    OpponentInfoConfig opponentInfoConfig;
+
+    @Mock
+    @Bind
+    OpponentInfoOverlay opponentInfoOverlay;
+
+    @Mock
+    @Bind
+    PlayerComparisonOverlay playerComparisonOverlay;
+
+    @Inject
+    OpponentInfoPlugin opponentInfoPlugin;
+
+    @Before
+    public void before()
+    {
+        Guice.createInjector(BoundFieldModule.of(this)).injectMembers(this);
+    }
+
+    /*
+     * Verify that the current opponent in multi is always the npc that the player is attacking
+     */
+    @Test
+    public void testAttackingEnemyInMulti() {
+        Player thePlayer = mock(Player.class), otherPlayer = mock(Player.class);
+        NPC aggro = mock(NPC.class), attacked = mock(NPC.class);
+        when(client.getLocalPlayer()).thenReturn(thePlayer);
+
+        opponentInfoPlugin.onInteractingChanged(new InteractingChanged(thePlayer, null));
+
+        // verify that there is currently no opponent
+        assertNull(opponentInfoPlugin.getLastOpponent());
+
+        // some npc attacks the player
+        InteractingChanged aggroEnemy = new InteractingChanged(aggro, thePlayer);
+        opponentInfoPlugin.onInteractingChanged(aggroEnemy);
+
+        // verify that the current opponent is the aggressive npc
+        assertSame(opponentInfoPlugin.getLastOpponent(), aggro);
```

Comment:
FYI, the argument order for assertions is `assertSame(expected, actual)`, so every assertion has the parameters in reverse order.
```suggestion
        assertSame(opponentInfoPlugin.getLastOpponent(), aggro);
```